dbms/src/Storages/DeltaMerge/SegmentReadTaskPool.h

```diff
@@ -167,6 +167,8 @@ class SegmentReadTaskPool : private boost::noncopyable
 
     std::once_flag & addToSchedulerFlag() { return add_to_scheduler; }
 
+    void registerPipeTask(TaskPtr && task) { q.registerPipeTask(std::move(task)); }
```

Comment:
How about letting `SegmentReadTaskPool` inherit from `NotifyFuture`? If so, `ScanNotifyFuture` is not needed anymore.

Reply:
good point! updated.